qa/tasks/kafka.py

```diff
@@ -0,0 +1,401 @@
+    finally:
+        pass
+
+@contextlib.contextmanager
+def configure(ctx, config):
+    """
+    Configure the bucket notifications tests for the kafka endpoint.
+    This includes writing in the local configuration file.
+    """
+    assert isinstance(config, dict)
+    log.info('Configuring bucket-notifications-tests...')
+    testdir = teuthology.get_testdir(ctx)
+
+    for client, properties in config['clients'].items():
+        bntests_conf = config['bntests_conf'][client]
+        bntests_conf['DEFAULT']['num_zonegroups'] = '1'
+        bntests_conf['DEFAULT']['num_zones'] = '1'
+        bntests_conf['DEFAULT']['num_ps_zones'] = '0'
+        bntests_conf['DEFAULT']['gateways_per_zone'] = '1'
+        bntests_conf['DEFAULT']['no_bootstrap'] = False
+        bntests_conf['DEFAULT']['log_level'] = '5'
+
+        (remote,) = ctx.cluster.only(client).remotes.keys()
+
+        conf_fp = BytesIO()
+        bntests_conf.write(conf_fp)
+
+        remote.write_file(
+            path='{tdir}/ceph/test_multi.{client}.conf'.format(tdir=testdir,client=client),
+            data=conf_fp.getvalue(),
+            )
+
+    try:
+        yield
+    finally:
+        log.info('Removing test_multi.conf file...')
+        for client, properties in config['clients'].items():
+            (remote,) = ctx.cluster.only(client).remotes.keys()
+            remote.run(
+                 args=['rm', '-f',
+                       '{tdir}/ceph/test_multi.{client}.conf'.format(tdir=testdir,client=client),
+                 ],
+                 )
+
+@contextlib.contextmanager
+def configures(ctx, config):
+    assert isinstance(config, dict)
+    log.info('Configuring bucket-notifications-tests...')
+    testdir = teuthology.get_testdir(ctx)
+    for client in config['clients']:
+        (remote,) = ctx.cluster.only(client).remotes.keys()
+        remote.run(
+            args=[
+                'cd', 
+                '{tdir}/ceph/src/test/rgw/bucket_notification'.format(tdir=testdir),
+                run.Raw('&&'),
+                './bootstrap',
+                ],
+            )
+
+
+@contextlib.contextmanager
+def run_tests(ctx, config):
+    """
+    Run the bucket notifications tests after everything is set up.
+    :param ctx: Context passed to task
+    :param config: specific configuration information
+    """
+    assert isinstance(config, dict)
+    log.info('Running bucket-notifications-tests...')
+    testdir = teuthology.get_testdir(ctx)
+    for client, client_config in config.items():
+        (remote,) = ctx.cluster.only(client).remotes.keys()
+
+        '''
+        args = [
+            'KAFKA_DIR={dir}'.format(dir=get_kafka_dir(ctx)),
+            ]
+        args += [
+            'RGW_MULTI_TEST_CONF={tdir}/ceph/test_multi.{client}.conf'.format(tdir=testdir,client=client)
+            ]
+        args += [
+            'python3', '-m', 'nose',
+            '-w',
+            '{tdir}/ceph/src/test/rgw/rgw_multi'.format(tdir=testdir),
```

Comment:
don't run the multisite test

Reply:
I've commented them out but yet need to remove the function entirely

Reply:
please remove all multisite/pubsub stuff